File: src/functions/willV2/llm.py

from langchain_core.messages import HumanMessage, SystemMessage
from langchain_openai import ChatOpenAI

from constants import SETTINGS
from functions.willV2.system import system_message


class WillV2:
  ''' Class to represent the WillV2 language model (will shitspeare)
  
  '''
  def __init__(self):
    ''' Constructor for the WillV2 class
    '''
    self.model = self.start_model()


  def start_model(self):
    ''' Function to start initialise the fine-tuned language model
  
    Returns:
      openai: The language model
    '''
    model = ChatOpenAI(
      api_key=SETTINGS.openai_api_key.get_secret_value(),
      model="ft:gpt-3.5-turbo-0125:aleios:willv2:94uyrpGH",
      temperature=0.75,
      max_tokens=1500,
    )

    return model
  

  def invoke_ft(self, text: str) -> str:
    ''' Function to invoke the language model
    '''    
    
    messages = [
      SystemMessage(
        content=system_message
      ),
      HumanMessage(
        content=text
      )
    ]
    
    response = self.model.invoke(messages).content

    return response
  

File: apps/backend/lib/resources/functions/willV2/llm.ts
import { getEnvVariable } from "@article-gpt/helpers";
import { ChatOpenAI } from "langchain/chat_models/openai";
import { HumanMessage, SystemMessage } from "langchain/schema";
import { systemPrompt } from "./prompt";

export const invoke = async (message: string): Promise<string> => {
  
  const model = new ChatOpenAI({
    openAIApiKey: getEnvVariable("OPENAI_API_KEY"),
    modelName: "ft:gpt-3.5-turbo-0125:aleios:willv2:94uyrpGH",
    temperature: 0.75,
    maxTokens: 1500,
  });

  const messages: (HumanMessage | SystemMessage)[] = [
    new SystemMessage(systemPrompt),
    new HumanMessage(message),
  ];

  const response = await model.invoke(messages);

  const content = response.content;

  return content as string;
};
